src/components/legend.js

```diff
@@ -83,7 +86,7 @@ let Legend = class Legend extends React.Component {
     };
 
     return (
-      <div className="bg-white absolute bottom left ml12 mb36 py12 px12 shadow-darken10 round z1 w300">
+      <div className={`legend ${this.props.sidebarOpen ? "resize" : ""} bg-white absolute bottom left mb36 shadow-darken10 round z1 w300`}>
```

Comment:
Rather than making multiple components subscribe to `sidebarOpen` state changes, it might be easier to have a higher level component like Main watch `sidebarOpen`'s value and toggle a class on its div. Then all of the other components could rely on one CSS rule to override their position, e.g.

```css
.main.sidebarOpen .legend, .main.sidebarOpen .toggleButton {
    left: 320px;
}
```

You'd need to give Main the .main class and give Legend actual CSS instead of relying on Assembly's classes, which are weirdly aggressive with !important.